insights/parsers/softnet_stat.py

```diff
@@ -0,0 +1,100 @@
+"""
+softnet_stat - /proc/net/softnet_stat
+=====================================
+
+This parser parses the stats from network devices. These stats includes events
+per cpu(in row), number of packets processed i.e packet_process (first column),
+number of packet drops packet_drops (second column), time squeeze i.e net_rx_action
+performed time_squeeze(third column),cpu collision i.e collision occur while obtaining
+device lock while transmitting cpu_collision packets (Ninth column), received_rps
+number of times cpu woken up received_rps (tenth column), number of times reached flow
+limit count flow_limit_count (eleventh column).
+
+    The typical contents of ``cat /proc/net/softnet_stat`` file is as below::
+
+        00008e78 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        000040ee 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        0001608c 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        0000372f 00000001 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+
+    Examples:
+        >>> SOFTNET_STATS = '''
+        ... 00008e78 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ... 000040ee 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ... 0001608c 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ... 0000372f 00000001 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ...'''.strip()
+        >>> softnet_obj = shared[SoftNetStats]
+        >>> softnet_obj.is_packet_drops()
+        True
+        >>> softnet_obj.get_cpu_instaces()
+        4
+        >>> softnet_obj.per_cpu_nstat('packet_drops')
+        {'cpu2': '00000000', 'cpu3': '00000001', 'cpu0': '00000000', 'cpu1': '00000000'}
+"""
+
+from .. import Parser, parser
+
+
+@parser("softnet_stat")
+class SoftNetStats(Parser):
+    """Parses ``/proc/net/softnet_stat`` file contains"""
+
+    def parse_content(self, contents):
+        cpu = 0
+        data = {}
+        for line in contents:
+            stats = line.split(' ')
+
+            data['cpu' + str(cpu)] = {'packet_process': stats[0],
```

Comment:
I'm not sure why this is a dictionary rather than an array based on CPU number, since CPU numbers will always be contiguous.  It also makes it easy to do array comprehensions.

Reply:
Made it this way because based on the rows in file it doesn't tell what the row implies .